core/shared/src/main/scala/zio/FiberRef.scala

```diff
@@ -103,10 +109,92 @@ final class FiberRef[A](private[zio] val initial: A, private[zio] val combine: (
     (result, result)
   }
 
+  /**
+   * Returns a handle that can be used to interact with this `FiberRef` from side effecting code.
+   *
+   * This feature is meant to be used for integration with side effecting code, that needs to access fiber specific data,
+   * like MDC contexts and the like.
+   *
+   * Important: Enable `Platform.fiberContextPropagation` if you want to use this.
+   */
+  final def unsafeHandle: UIO[UnsafeHandle[A]] =
```

Comment:
I think we need to change this to `UIO[ThreadLocal[A]]`. We can implement all methods:

 - `get` (`unsafeGet` in `UnsafeHandle`)
 - `set` (`unsafeSet` in `UnsafeHandle`)
 - `initialValue` (the private initial value in `FiberRef`)
 - `remove` (same as `set(null)`?)

`ThreadLocal` is widely understood by existing Java / Scala developers who would be most likely to use this feature. We can piggyback atop their expectations of how it should work.

Reply:
Fair enough, I'll look into that over the weekend

Reply:
I've just dropped `FiberRef.unsafeHandle` in favour of `FiberRef.unsafeAsThreadLocal` and adapted the implementation as outlined by you above.